src/workflow_tasks.rs

```diff
@@ -383,14 +403,15 @@ impl WorkflowTaskManager {
             );
             return FailedActivationOutcome::NoReport;
         };
+        self.cache_manager.write().evict(run_id);
```

Comment:
Why does this line need to be here? The below call to `evict_run` should be sufficient?

Reply:
Yeah, no need to do it here as it's post activation completion.